Add details on installation (#819)
ambanum/OpenTermsArchive: README.md

### Installing

> - This module is built with [Node](https://nodejs.org/en/). You will need to [install Node >= v14.x](https://nodejs.org/en/download/) to run it.

Reviewer: > I was given feedback through another channel that we actually needed Node 16. Just to check this information, did you try with Node 14 or 16? 🙂

Author: I currently have Node v16.14.0

> - When refering to the root folder, it means the folder where you will be `git pull`ing everything.

Reviewer: This is ambiguous: “root” has a specific meaning on the filesystem. I would prefer that we avoid naming this folder. If we do need it though, I believe that “work folder” or “base folder” would be more appropriate 🙂

#### Declarations repository

> 1. Locally clone your declaration repository, e.g., `git@github.com:OpenTermsArchive/contrib-declarations.git`.

Reviewer: ```suggestion
1. Locally clone your declarations repository, e.g., `git@github.com:OpenTermsArchive/contrib-declarations.git`.
```

> 2. Go into your folder and initialize it, e.g., `cd contrib-declarations; npm install`.

Reviewer: ```suggestion
2. Go into the cloned folder and install dependencies: `cd contrib-declarations; npm install`.
```

3. You can now modify your declarations in the `./declarations/` folder, following [these instructions](https://github.com/OpenTermsArchive/contrib-declarations/blob/main/CONTRIBUTING.md).

> 4. When you want to test:

Reviewer: I feel this part, while useful, duplicates information already present in the CONTRIBUTING referenced above. We could maybe integrate the improvements given here into that file, and remove the part here.

Author: No problem - let me know!

Reviewer: Let's move it to https://github.com/OpenTermsArchive/contrib-declarations/blob/main/CONTRIBUTING.md then 🙂

    - If you want to test every declaration, run `npm test`.

>     - If you want to test a specific declaration, run `npm test <name of the service>`, e.g., `npm test HER`.

Reviewer: We use `$`-prefixed syntax for variables in the rest of the README, let's keep this homogenous 🙂 
```suggestion
    - If you want to test a specific declaration, run `npm test $serviceId`, e.g., `npm test HER`.
```

>     - If you want to only test the integrity of a specific declaration, run `npm test -- --schema-only <name of the service>`, e.g., `npm test -- --schema-only HER`.

Reviewer: “integrity” is ambiguous.
```suggestion
    - If you want to have faster feedback on the structure of a specific declaration, run `npm test -- --schema-only $serviceId`, e.g., `npm test -- --schema-only HER`.
```

5. Once you have done that, if you have any error, it will be prompted and detailed at the end of the test.
    - E.g., `InaccessibleContentError`: Your selector is wrong and should be fixed.
    - E.g., `TypeError`: The file declaration is invalid.

>     - E.g., if you have a weird error, you may want to contact OTA, if may be a bug.

Reviewer: I am not sure of the added value of this part 🙂

Author: I mainly put that in our side to get more insights about errors - let me know if I should rather remove that part.

As previously discussed, this may be too many details for an internal usage indeed.

Reviewer: If this was not clear enough and granted writing documentation, I believe this should lead to improving the test failure messages instead 🙂

I would prefer that we remove this from the doc. If you feel like trying to improve the logging messages, feel free to open a PR improving that file: https://github.com/ambanum/OpenTermsArchive/blob/master/scripts/validation/validate.js 👍

##### Note: Testing

Testing works with multiple tests:

- Checking the validity of the file.
- URL is correct.
- The content is correctly gathered.
- Quick diff between two "same" contents, to check for noise.
- At least 100 chars.

> - Etc.

Reviewer: Describing the whole test suite here increases the risk of documentation rot. I would prefer if we just gave one or two examples between parentheses rather than providing this list, so we don't have to remember maintaining it 🙂
